File: android/src/main/java/com/app/yinshi_plugin/EventStreamHander.java
package com.app.yinshi_plugin;

import io.flutter.plugin.common.EventChannel;

public class EventStreamHander implements EventChannel.StreamHandler {

    private EventChannel.EventSink eventSink;
    @Override
    public void onListen(Object arguments, EventChannel.EventSink events) {
        eventSink = events;
    }

    @Override
    public void onCancel(Object arguments) {
        eventSink = null;
    }

    //发送event
    public void sendEvent(Object event) {
        eventSink.success(event);
    }
}

File: ios/Classes/EventStreamHander.swift
//
//  EventStreamHander.swift
//  yinshi_plugin
//
//  Created by Peixue Yan on 2020/8/10.
//

import Foundation

class EventStreamHander: FlutterStreamHandler {
    private var eventSink:FlutterEventSink? = nil
       func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
           eventSink = events
           return nil
       }
       
       func onCancel(withArguments arguments: Any?) -> FlutterError? {
           eventSink = nil
           return nil
       }
       
       //发送event
       public func sendEvent(event:Any) {
        eventSink?(event)
       }
}
